pandas/core/internals/managers.py

```diff
@@ -696,8 +697,9 @@ def get_bool_data(self, copy: bool = False) -> "BlockManager":
         copy : bool, default False
             Whether to copy the blocks
         """
-        self._consolidate_inplace()
-        return self._combine([b for b in self.blocks if b.is_bool], copy)
+        # Note: use is_bool_dtype instead of blk.is_bool to exclude
+        #  object-dtype blocks containing all-bool entries.
+        return self._combine([b for b in self.blocks if is_bool_dtype(b.dtype)], copy)
```

Comment:
this may actually be a bug, since it implies the following inconsistent behavior in master:

```
ser = pd.Series([True, False, True], dtype=object)
ser2 = pd.Series(["A", "B", "C"])

df = ser.to_frame("A")
>>> df._get_bool_data()
       A
0   True
1  False
2   True

df["B"] = ser2
>>> df._get_bool_data()
Empty DataFrame
Columns: []
Index: [0, 1, 2]
```

adding columns shouldnt make get_bool_data smaller.

Reply:
can you add a test for this

Reply:
test added